tlsn/tlsn-notary-client/src/client.rs

```diff
@@ -0,0 +1,412 @@
+                        Some("Failed to attach http client to notary socket".to_string()),
+                        Some(Box::new(err)),
+                    )
+                })?;
+
+        // Spawn the HTTP task to be run concurrently
+        let notary_connection_task = tokio::spawn(notary_connection.without_shutdown());
+
+        // Build the HTTP request to configure notarization
+        let configuration_request_payload = serde_json::to_string(&NotarizationSessionRequest {
+            client_type: ClientType::Tcp,
+            max_sent_data: Some(notarization_request.max_sent_data),
+            max_recv_data: Some(notarization_request.max_recv_data),
+        })
+        .map_err(|err| {
+            ClientError::new(
+                ErrorKind::Configuration,
+                Some("Failed to serialise http request for configuration".to_string()),
+                Some(Box::new(err)),
+            )
+        })?;
+
+        let mut configuration_request_builder = Request::builder()
+            .uri(format!(
+                "{http_scheme}://{}:{}/session",
+                self.host, self.port
+            ))
+            .method("POST")
+            .header("Host", &self.host)
+            // Need to specify application/json for axum to parse it as json
+            .header("Content-Type", "application/json");
+
+        if let Some(api_key) = &self.api_key {
+            configuration_request_builder =
+                configuration_request_builder.header("Authorization", api_key);
+        }
+
+        let configuration_request = configuration_request_builder
+            .body(Either::Left(Full::new(Bytes::from(
+                configuration_request_payload,
+            ))))
+            .map_err(|err| {
+                ClientError::new(
+                    ErrorKind::Configuration,
+                    Some("Failed to build http request for configuration".to_string()),
+                    Some(Box::new(err)),
+                )
+            })?;
+
+        #[cfg(feature = "tracing")]
+        debug!("Sending configuration request: {:?}", configuration_request);
+
+        let configuration_response = notary_request_sender
+            .send_request(configuration_request)
+            .await
+            .map_err(|err| {
+                ClientError::new(
+                    ErrorKind::Configuration,
+                    Some("Failed to send http request for configuration".to_string()),
+                    Some(Box::new(err)),
+                )
+            })?;
+
+        #[cfg(feature = "tracing")]
+        debug!("Sent configuration request");
+
+        if configuration_response.status() != StatusCode::OK {
+            return Err(ClientError::new(
+                ErrorKind::Configuration,
+                Some(format!(
+                    "Configuration response is not OK: {:?}",
+                    configuration_response
+                )),
+                None,
+            ));
```

Comment:
This should be `Internal`? same like when notarization response isn't `SWITCHING_PROTOCOL` below

Reply:
Actually this should be `Configuration`, since the most likely reason it ll err is cause user pass in invalid max_sent or max_recv (that exceeds that max set in notary server) or invalid API key if server whitelists. What do you think?